### ISSUE TITLE
Central error while changing rule tag in review stage

### ISSUE THREAD

**Issue by roheline** [cert-ee/s4a#6]
```
Apr 17 10:15:47 central pm2[4557]: 10:15:47 0|s4a-cent | [error] [rule_draft.publish] Cannot read property 'added' of undefined
```

No errors occur while adding single rule.

**Comment by roheline** [comment]
Other rule changes made in review stage are not causing errors but are also not saved.

**Comment by tavipoldma** [comment]
and the testing branch, seems ok?

**Comment by roheline** [comment]
Bug still exists while adding a tag to a new custom rule in review stage. 

**Central version:**
```
server
0.1.97
client
0.1.104
main
0.1.120
```

**syslog**
```
Nov 26 12:50:55 central pm2[10881]: 12:50:55 0|s4a-central-api-server  | [info] [rule_draft.more] start
Nov 26 12:50:55 central pm2[10881]: 12:50:55 0|s4a-central-api-server  | [info] [rule_draft.more] 10000000 new rule
Nov 26 12:50:55 central pm2[10881]: 12:50:55 0|s4a-central-api-server  | [info] [rule_draft.more] 10000000 look for duplicate sid
Nov 26 12:50:55 central pm2[10881]: 12:50:55 0|s4a-central-api-server  | [info] [rule_draft.more] [] []
Nov 26 12:50:55 central pm2[10881]: 12:50:55 0|s4a-central-api-server  | [info] [rule_draft.more] 10000000 no duplicate, continue
Nov 26 12:50:55 central pm2[10881]: 12:50:55 0|s4a-central-api-server  | [info] [rule_draft.more] 10000000 new custom draft created
Nov 26 12:50:55 central pm2[10881]: 12:50:55 0|s4a-central-api-server  | [info] [rule_draft.more] new done
Nov 26 12:51:10 central pm2[10881]: 12:51:10 0|s4a-central-api-server  | [info] [rule_draft.publish] start
Nov 26 12:51:10 central pm2[10881]: 12:51:10 0|s4a-central-api-server  | [info] [rule_draft.publish] find drafts
Nov 26 12:51:10 central pm2[10881]: 12:51:10 0|s4a-central-api-server  | [info] [rule_draft.publish] 10000000 loop
Nov 26 12:51:10 central pm2[10881]: 12:51:10 0|s4a-central-api-server  | [info] [rule_draft.publish] 10000000 new rule
Nov 26 12:51:10 central pm2[10881]: 12:51:10 0|s4a-central-api-server  | [info] [rule_draft.publish] 10000000 create rule
Nov 26 12:51:10 central pm2[10881]: 12:51:10 0|s4a-central-api-server  | [info] [rule_draft.publish] 10000000 published rule 5bfbd01ee0a83e2af2a51092
Nov 26 12:51:10 central pm2[10881]: 12:51:10 0|s4a-central-api-server  | [info] [rule_draft.publish] 10000000 tags len 1
Nov 26 12:51:10 central pm2[10881]: 12:51:10 0|s4a-central-api-server  | [info] [rule_draft.publish] 10000000 remove draft
Nov 26 12:51:10 central pm2[10881]: 12:51:10 0|s4a-central-api-server  | [info] [rule_draft.publish] 10000000 draft removed
```
